src/main/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo.java

```diff
@@ -624,51 +625,104 @@ private int countViolations( XmlPullParser xpp )
                 file = PathTool.getRelativeFilePath( basedir, xpp.getAttributeValue( "", "name" ) );
-                //file = file.substring( file.lastIndexOf( File.separatorChar ) + 1 );
+                continue;
             }
-            else if ( "error".equals( xpp.getName() ) )
+            else if ( ! "error".equals( xpp.getName() ) )
             {
-                String severity = xpp.getAttributeValue( "", "severity" );
+                continue;
+            }
 
-                if ( !isViolation( severity ) )
-                {
-                    continue;
-                }
+            String severity = xpp.getAttributeValue( "", "severity" );
+            String source = xpp.getAttributeValue( "", "source" );
+            String line = xpp.getAttributeValue( "", "line" );
+            /* Nullable */
+            String column = xpp.getAttributeValue( "", "column" );
+            String message = xpp.getAttributeValue( "", "message" );
+            String rule = RuleUtil.getName( source );
+            String category = RuleUtil.getCategory( source );
+
+            Violation violation = new Violation(
+                source,
+                file,
+                line,
+                severity,
+                message,
+                rule,
+                category
+            );
+            if ( column != null )
+            {
+                violation.setColumn( column );
+            }
 
-                String source = xpp.getAttributeValue( "", "source" );
+            violations.add( violation );
+        }
 
-                if ( ignore( ignores, source ) )
-                {
-                    ignoreCount++;
-                }
-                else
-                {
-                    count++;
+        return Collections.unmodifiableList( violations );
+    }
 
-                    if ( logViolationsToConsole )
-                    {
-                        String line = xpp.getAttributeValue( "", "line" );
-                        String column = xpp.getAttributeValue( "", "column" );
-                        String message = xpp.getAttributeValue( "", "message" );
-                        String rule = RuleUtil.getName( source );
-                        String category = RuleUtil.getCategory( source );
-
-                        log( severity, file + ":[" + line + ( ( column == null ) ? "" : ( ',' + column ) ) + "] ("
-                            + category + ") " + rule + ": " + message );
-                    }
-                }
+    private int countViolations( List<Violation> violations )
+    {
+        List<RuleUtil.Matcher> ignores = violationIgnore == null ? Collections.<RuleUtil.Matcher>emptyList()
+            : RuleUtil.parseMatchers( violationIgnore.split( "," ) );
+
+        int ignored = 0;
+
+        List<Violation> actualViolations = new ArrayList<>();
+
+        for ( Violation violation : violations )
+        {
+            if ( ! isViolation( violation.getSeverity() ) )
```

Comment:
This guard pattern and the use of continue is uncommon at best. This one could easily and I argue more clearly be written with if-else if-else and no continues.

Reply:
Interesting. I thought all `else`s are not much used anymore, because a guard statement is much clearer.

* https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a
* https://stackoverflow.com/questions/4887212/shall-i-use-guard-clause-and-try-to-avoid-else-clause
* https://refactoring.guru/replace-nested-conditional-with-guard-clauses


Maybe implement a checkstyle check "avoid use of guard statements"? At least the code went through fine. I do not see anything uncommon in this, and I agree with the SO answer that this is really just a matter of taste. That said, do all commits need to suit your taste or is there a convention you did not mention?

Reply:
Again, I read lots of code, and I have never seen anyone try this before.  The last one you cite is something very different. The first two suggest this might be used, but neither is normative or widely accepted in the community. If you do something unusual, there should be a strong reason for it. Anything that makes the reader stop and think about why the code is the way it is, is not ideal.

Reply:
Thats exactly the other way for me. How should I know this is not accepted by the community as a new committer? And then you implicitly agree this is only esthetics.

Please feel free to change it yourself, because it is not a functional change. Also, it embraces more indentation when there will be more options, and I do not want to be the source of that excessive nesting.